code/game/machinery/autolathe.dm

```diff
@@ -135,10 +135,10 @@
 			use_power(max(500,M.last_amount_inserted/10))
 		else
 			var/obj/item/stack/S = lit
-			var/list/initmats = initial(S.materials)
-			if (initmats[MAT_METAL])
+			var/list/mats = S.materials

```

Comment:
S is a typepath

Reply:
Oh, i guess that explains the attempt then. I'll have to look again.